src/no-type-assertion-as.ts

```diff
@@ -0,0 +1,60 @@
+// no-type-assertion-as.ts
+
+/*
+ * Copyright (c) 2021-2025 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import { AST_NODE_TYPES, ESLintUtils, TSESTree } from '@typescript-eslint/utils';
+
+export const ruleId = 'no-as-type-assertion';
+const NO_AS_TYPE_ASSERTION = 'NO_AS_TYPE_ASSERTION';
+
+const createRule = ESLintUtils.RuleCreator((name) => name);
+
+const rule: ESLintUtils.RuleModule<typeof NO_AS_TYPE_ASSERTION> = createRule({
+  name: ruleId,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow the use of `as` type assertions and suggest using `satisfies` instead',
+    },
+    schema: [],
+    messages: {
+      [NO_AS_TYPE_ASSERTION]: 'Avoid using `as` type assertions. Use `satisfies` instead.',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      TSAsExpression(node: TSESTree.TSAsExpression) {
+        const isUnknownUnionOrIntersection =
+          (node.typeAnnotation.type === AST_NODE_TYPES.TSUnionType ||
+            node.typeAnnotation.type === AST_NODE_TYPES.TSIntersectionType) &&
+          node.typeAnnotation.types.some((type) => type.type === AST_NODE_TYPES.TSUnknownKeyword);
+
+        if (
```

Comment:
this is a complex bit of logic I don't quite understand, if there was just one `context.report()` it would help.  also maybe some more `isSomethingSomething` booleans to help readability.